sim/make/cpp-lint.mk

```diff
@@ -24,6 +24,7 @@ clang_tidy_files := $(shell \
 clang_tidy_flags :=\
 	-I$(firesim_base_dir)/midas/src/main/cc \
 	-I$(firesim_base_dir)/firesim-lib/src/main/cc \
+	-I$(firesim_base_dir)/src/main/cc/midasexamples \
```

Comment:
To settle some more of the style issues, I would suggest enabling `clang-tidy` on the header. [Here](https://github.com/firesim/firesim/blob/71b8eb9c137deccb104b8e5044221eca36203a7a/sim/make/cpp-lint.mk#L15), we could filter out the `.cpp` and nested stuff while still validating the header. The problem is with the fact that `clang-tidy` needs the full set of dependencies which are slow to install. If we upgraded libelf/libdwarf, we could install them through the system packages and we'd be able to run clang-tidy.